cmdLine/src/main/scala/topgun/cmdline/FileParser.scala

```diff
@@ -12,63 +12,95 @@ class FileParser(file: File, cmdLine: JfrParseCommandLine, totals: Totals, confi
+              else throw new InvalidJfrSettingException(s"failed [file]${file.getName} Reason: 'Allocation outside TLAB' event not enabled")
+
+            case RecordedEventTypeIdMapping.MethodProfilingSample =>
+              if(value.toBoolean) FoundMethodProfilingSampleEnabled=true
+              else throw new InvalidJfrSettingException(s"failed [file]${file.getName} Reason: 'Method Profiling Sample' event not enabled")
+
+            case _ =>
+          }
+        }
+      }
+
+      def notDoneFetchingInfo():Boolean = {
+        !FoundAllocationInNewTlabEnabled ||  !FoundAllocationOutsideTlabEnabled || !FoundMethodProfilingSampleEnabled ||
+          isWindow.isEmpty || classPath.isEmpty
+      }
+
       val recordingFile = new RecordingFile(file.toPath)
-      //      for (event <- view.asScala) {
-      while (recordingFile.hasMoreEvents){
+      while(recordingFile.hasMoreEvents && notDoneFetchingInfo()){
         val event = recordingFile.readEvent()
-        event.getEventType.getLabel match {
-          case "Allocation in new TLAB" => allocationInNewTlab=true; allocation(event, true)
-          case "Allocation outside TLAB" => allocationOutsideTlab=true; allocation(event, false)
-          case "Method Profiling Sample" => methodProfilingSample=true; cpu(event)
-          case e =>
-            totals.ignoreEvent(e)
-
-          //            case "GC Configuration" =>
-          //              println(event)
-          //            case "GC TLAB Configuration" =>
-          //              handleTlabConfiguration(event)
+        event.getEventType.getId match {
+
+          case RecordedEventTypeIdMapping.InitialSystemProperty if classPath.isEmpty => if(event.getValue("key").toString.contains("java.class.path"))
+            classPath = event.getValue("value").toString
+
+          case RecordedEventTypeIdMapping.RecordingSetting => extractJfrSettings(event)
+
+          case RecordedEventTypeIdMapping.OSInformation if isWindow.isEmpty =>
+            isWindow = Some(event.getValue("osVersion").toString.toLowerCase.contains("win"))
+
           case _ =>
         }
-        totals.totalEvents.incrementAndGet()
       }
-
-    } catch {
-      case e: Exception => e.printStackTrace()
+      recordingFile.close()
+      ClassLoaderFactory.addIfDoesNotExist(classPath,if (isWindow.get) ";" else ":" )
     }
 
-    def foundRequiredEvents(): Unit = {
-      if(!allocationInNewTlab){
-        throw new JfrEventNotFoundException(s"failed [file]${file.getName} Reason: 'Allocation in new TLAB' event not found")
-      }
-      if(!allocationOutsideTlab){
-        throw new JfrEventNotFoundException(s"failed [file]${file.getName} Reason: 'Allocation outside TLAB' event not found")
-      }
-      if(!methodProfilingSample){
-        throw new JfrEventNotFoundException(s"failed [file]${file.getName} Reason: 'Method Profiling Sample' event not found")
-      }
-    }
+    preProcessing()
```

Comment:
i think you want `val classPath = preProcesing()`
change the return type, remove the global var